internal/testingx/pubsub.go

```diff
@@ -0,0 +1,97 @@
+package testingx
+
+import (
+	"context"
+
+	"go.infratographer.com/permissions-api/pkg/permissions"
+	"go.infratographer.com/x/events"
+	"go.infratographer.com/x/gidx"
+)
+
+// TestPublisherMethod is a type for the methods of the TestPublisher
+type TestPublisherMethod string
+
+const (
+	// TestPublisherMethodCreate is the method name for CreateAuthRelationships
+	TestPublisherMethodCreate TestPublisherMethod = "CreateAuthRelationships"
+	// TestPublisherMethodDelete is the method name for DeleteAuthRelationships
+	TestPublisherMethodDelete TestPublisherMethod = "DeleteAuthRelationships"
+)
+
+// TestPublisherCalledWith records the arguments passed to the TestPublisher
+type TestPublisherCalledWith struct {
```

Comment:
Is there a reason not to use the existing mockpermissions? https://github.com/infratographer/permissions-api/blob/main/pkg/permissions/mockpermissions/permissions.go